Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -364,6 +366,10 @@ open class RouteController: NSObject {
     @objc func restoreToOnline(_ notification: Notification) {
         updateNavigator(with: self.routeProgress, completion: nil)
     }
+
+    func isValidNavigationStatus(_ status: NavigationStatus) -> Bool {
+        return routeProgress.currentLegProgress.leg.steps.indices.contains(Int(status.stepIndex))
```

Comment:
Would be good to have telemetry event for the cases when navigation status isn't valid.

Reply:
Should this also check for `NavigationStatus.routeState` being `invalid`? I’ve often had to instruct developers to set a breakpoint in `update(to:)` to check the route state, to narrow an issue down to the navigator rejecting a route or otherwise getting into a bad state. Aside from telemetry, this would be a useful piece of information to somehow bubble up to application code via a notification or delegate method. In any case, it would be fine to break out that change as tail work.

Reply:
That's a good question. I think I was afraid that we lose locations and it seems more invasive change that is out of scope for this PR. Even this change is, but it doesn't hurt and just removes crashes. I will cut a follow-up ticket to iterate on this problem.